contrib/mod_sql_postgres.c

```diff
@@ -1626,27 +1643,41 @@ static void sql_postgres_mod_unload_ev(const void *event_data,
 /* Initialization routines
  */
 
-static int sql_postgres_init(void) {
-
-  /* Register listeners for the load and unload events. */
-  pr_event_register(&sql_postgres_module, "core.module-load",
-    sql_postgres_mod_load_ev, NULL);
-  pr_event_register(&sql_postgres_module, "core.module-unload",
-    sql_postgres_mod_unload_ev, NULL);
-
+static void sql_postgres_ssl_init(void) {
 #ifdef HAVE_POSTGRES_PQINITOPENSSL
+  int init_ssl = TRUE, init_crypto = TRUE;
+
   /* If any of the OpenSSL-using modules are loaded, tell Postgres to NOT
-   * initialize OpenSSL itself.
+   * initialize OpenSSL itself.  Note that there are nuances to this; some
+   * of the other modules may only use the crypto libs.
    */
   if (pr_module_exists("mod_auth_otp.c") == TRUE ||
       pr_module_exists("mod_digest.c") == TRUE ||
-      pr_module_exists("mod_tls.c") == TRUE ||
       pr_module_exists("mod_sftp.c") == TRUE ||
       pr_module_exists("mod_sql_passwd.c") == TRUE) {
-    PQinitOpenSSL(0, 0);
+    init_crypto = FALSE;
+  }
+
+  if (pr_module_exists("mod_tls.c") == TRUE) {
+    init_ssl = FALSE;
+    init_crypto = FALSE;
   }
+
+  pr_trace_msg(trace_channel, 18,
+    "telling Postgres about OpenSSL initialization: ssl = %s, crypto = %s",
+    init_ssl ? "yes" : "no", init_crypto ? "yes" : "no");
+  PQinitOpenSSL(init_ssl, init_crypto);
```

Comment:
The root cause of:

    could not create SSL context: SSL error code 336236705

is that `mod_sql_postgres` was not informing the Postgres client library of what needed to be initialized properly, OpenSSL-wise.  And since OpenSSL was not properly initialized for creating SSL contexts (needed for SSL sessions), the "could not create SSL context" error was happening.

The above fixes this.